iter (and utter) weirdness

The following code doesn't compile:
```rust
use nonempty::NonEmpty;

fn main() {
    let mut l: NonEmpty<u32> = NonEmpty::new(42);

    assert_eq!(l.first(), &42);

    l.push(36);
    l.push(58);

    for i in l.iter() {
        println!("{}", i)
    }
}
```

and results in:
```
error[E0277]: `<impl std::iter::Iterator as std::iter::Iterator>::Item` doesn't implement `std::fmt::Display`
  --> src/main.rs:12:24
   |
12 |         println!("{}", i)
   |                        ^ `<impl std::iter::Iterator as std::iter::Iterator>::Item` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `<impl std::iter::Iterator as std::iter::Iterator>::Item`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required by `std::fmt::Display::fmt`

error: aborting due to previous error
```

So something about the fact that you're returning an `Iterator` from `iter` seems to be causing issues in some way.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -82,7 +82,21 @@ impl<T> NonEmpty<T> {
         self.1.truncate(len - 1);
     }
 
-    pub fn iter<'a>(&'a self) -> impl Iterator + 'a {
+    /// ```
+    /// use nonempty::NonEmpty;
+    ///
+    /// let mut l = NonEmpty::new(42);
+    /// l.push(36);
+    /// l.push(58);
+    ///
+    /// let mut l_iter = l.iter();
+    ///
+    /// assert_eq!(l_iter.next(), Some(&42));
+    /// assert_eq!(l_iter.next(), Some(&36));
+    /// assert_eq!(l_iter.next(), Some(&58));
+    /// assert_eq!(l_iter.next(), None);
+    /// ```
+    pub fn iter<'a>(&'a self) -> impl Iterator<Item = &T> + 'a {
         std::iter::once(&self.0).chain(self.1.iter())
     }
 }

